### PR TITLE
feat(prepare): Update source tracking implementation

### LINKED ISSUE #817: Changes to the SFDX CLI source tracking
The Force.com CLI team are changing the implementation of source tracking, namely `sourcePathInfos` will be deprecated and replaced by git-based source tracking; the location of source tracking files will change; and for multiple package directories, the packages will be merged and deployed in a single transaction. 

Some of these changes will impact sfpowerscripts source tracking functionality for scratch org pools, and need to be accommodated so that sfpowerscripts will continue to work.

For more information of the changes being introduced by the Force.com CLI team:
https://github.com/forcedotcom/cli/issues/1258

### PR DESCRIPTION
**Pull request by aly76**
Updates to the source tracking implementation for scratch org pools, to align with `force:source:beta` commands. 
Deprecate static resource in scratch orgs for storing source tracking files
Move pool implementation to core library

[Decision Record](https://github.com/Accenture/sfpowerscripts/blob/0aeb28376c72e3e2006030de3676090b19d7dce8/decision%20records/prepare/001-prepare-source-tracking.md)

#817

### CODE REVIEW

File: packages/core/src/sfdxwrappers/PushSourceImpl.ts [hunk 1]
```diff
@@ -6,7 +6,7 @@ export default class PushSourceImpl extends SFDXCommand {
     }
 
     public getSFDXCommand(): string {
-        return `sfdx force:source:push`;
+        return `sfdx force:source:beta:push`;
```
Comment (azlam-abdulsalam) [inline packages/core/src/sfdxwrappers/PushSourceImpl.ts]: What's the idea here?  should we get remove of this wrapper altogether?

Comment (aly76) [reply packages/core/src/sfdxwrappers/PushSourceImpl.ts]: Will remove beta once GA'd by Salesforce. Also looking to replace with source-deploy-retrieve API in the future.

**Comment by aly76** [comment]
Remaining tasks
- [x] Update SourceTrackingController for SFP CLI
- [X] Investigate whether maxRevision.json can be removed from static resource as well
- [x] Remove static resource
- [x] Remove beta from source commands when are GA

**Comment by azlam-abdulsalam** [comment]
@aly76  Can we remove duplication between sfp-cli and sfpowerscripts, Lets have a shared library or move to core

**Review by azlam-abdulsalam — APPROVED** [review]
okie

**Pull request merged**